### ISSUE TITLE
where is manual simple code in your final big code file 

### ISSUE THREAD

**Issue by Sandy4321** [maxbren/matrix-factorization-for-collaborative-filtering#2]
great material thanks
but
where is manual simple code in your final big code file 
# Create random initial latent matrices
U = np.random.uniform(-1, 1, size=(num_users, K))
V = np.random.uniform(-1, 1, size=(num_items, K))
N = num_users * num_items

# Training
for curr_epoch in range(num_epochs):
    # Reconstruct R_hat from latent factor matrices (linear operation)
    R_hat = U @ V.T  # @ means matrix multiplcation, .T means transpose
    # Calc MSE loss of this reconstruction
    loss = np.square(R - R_hat).mean()
    # Calc partial derivative of MSE with respect to U and V
    U_grad = -2./N * (R - R_hat)@V
    V_grad = -2./N * (R - R_hat)@U
    # Update U and V using their respective gradients and learning rate
    U -= lr * U_grad
    V -= lr * V_grad
